libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -27,25 +28,94 @@ CGroupsParser::CGroupsParser()
 {
 }
 
-ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
+std::string CGroupsParser::byteRegulation(const std::string settingValue,
+                                                        const int multiply)
+{
+    char *end;
+    long long limit_in_bytes = std::strtoll(settingValue.c_str(), &end, 10);
+    if (errno == ERANGE){
+        log_error() << settingValue << " is out of range";
+        throw LimitRangeError();
+    }
+
+    if (limit_in_bytes * multiply > std::numeric_limits<long long>::max()) {
```

Comment:
I think, since i am not assigning it it won't create overflow. The other way you propose can create floating comparison which i didn't like that much.